Mixpanel/Mixpanel.m

```diff
@@ -1122,6 +1126,7 @@ - (void)checkForDecideResponseWithCompletion:(void (^)(NSArray *surveys, NSArray
                     if (completion) {
                         completion(nil, nil, nil, nil);
                     }
+                    dispatch_semaphore_signal(semaphore);
```

Comment:
I think these `dispatch_semaphore_signal` should be in a separate PR so it could be merged with less risk. This definitely seems like a bug, as the semaphore lock will never go away if the device is offline during init.